src/common/com/intellij/plugins/haxe/model/HaxeCustomOperatorsModel.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2000-2013 JetBrains s.r.o.
+ * Copyright 2014-2015 AS3Boyan
+ * Copyright 2014-2014 Elias Ku
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.model;
+
+import com.intellij.plugins.haxe.util.UsefulPsiTreeUtil;
+import com.intellij.psi.impl.source.tree.java.PsiJavaTokenImpl;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class HaxeCustomOperatorsModel {
+  private final Map<String, HaxeCustomOperatorModel> operators = new LinkedHashMap<String, HaxeCustomOperatorModel>();
```

Comment:
This class is fairly heavyweight for how it's being used.  Do you have another use for it besides HaxeOperatorResolver._getBinaryOperatorResult()?

For that use case there is no need to pre-load all of the operators in a class and to create all of those objects just to go and check whether the operator exists.  Just a simple scan can tell you that information.  A static version of this.get(operator) would suffice and be a lot more efficient.